easybuild/tools/toolchain/toolchain.py

```diff
@@ -584,6 +588,31 @@ def _verify_toolchain(self):
             raise EasyBuildError("List of toolchain dependency modules and toolchain definition do not match "
                                  "(found %s vs expected %s)", self.toolchain_dep_mods, toolchain_definition)
 
+    def symlink_compilers(self, paths):
+        """
+        Create a symlink for each compiler to binary/script at specified path.
+
+        @param paths: dictionary containing mapping from types of caches (ccache, f90cache) to path to symlink to
+        """
+        tmpdir = tempfile.mkdtemp()
+        compilers = {}
+
+        if self.name == DUMMY_TOOLCHAIN_NAME:
+            compilers[CCACHE] = ['gcc', 'g++']
+            compilers[F90CACHE] = ['gfortran']
+        else:
+            compilers[CCACHE] = [self.COMPILER_CC, self.COMPILER_CXX]
+            compilers[F90CACHE] = [self.COMPILER_F77, self.COMPILER_F90, self.COMPILER_FC]
```

Comment:
this should be made more generic, since now `symlink_compilers` is specific to the compiler caching...

I think the argument passed to `symlink_compilers` should contain all information of which commands should be linked to which path, so something like:

```python
{
    'ccache': (`/path/to/ccache`, ['gcc', 'g++']),
    'f90cache': (`/path/to/f90cache`, ['gfortran']),
}
```

so constructing this should be done in the `if compiler_cache` in prepare, where we should call out to a separate function rather than blowing up the `prepare` method (especially since more stuff will be added there later for rpath too)